src/configs/svelte.ts

```diff
@@ -0,0 +1,93 @@
+import { ensurePackages, interopDefault } from '../utils'
+import type { FlatConfigItem, OptionsFiles, OptionsHasTypeScript, OptionsOverrides, OptionsStylistic } from '../types'
+import { GLOB_SVELTE } from '../globs'
+
+export async function svelte(
+  options: OptionsHasTypeScript & OptionsOverrides & OptionsStylistic & OptionsFiles = {},
+): Promise<FlatConfigItem[]> {
+  const {
+    files = [GLOB_SVELTE],
+    overrides = {},
+    stylistic = true,
+  } = options
+
+  const {
+    indent = 2,
+    quotes = 'single',
+  } = typeof stylistic === 'boolean' ? {} : stylistic
+
+  await ensurePackages([
+    'eslint-plugin-svelte',
+  ])
+
+  const [
+    pluginSvelte,
+    parserSvelte,
+  ] = await Promise.all([
+    interopDefault(import('eslint-plugin-svelte')),
+    interopDefault(import('svelte-eslint-parser')),
+  ] as const)
+
+  return [
+    {
+      name: 'antfu:svelte:setup',
+      plugins: {
+        svelte: pluginSvelte,
+      },
+    },
+    {
+      files,
+      languageOptions: {
+        parser: parserSvelte,
+        parserOptions: {
+          extraFileExtensions: ['.svelte'],
+          parser: options.typescript
+            ? await interopDefault(import('@typescript-eslint/parser')) as any
+            : null,
+        },
+      },
+      name: 'antfu:svelte:rules',
+      rules: {
+        'import/no-mutable-exports': 'off',
+        'no-undef': 'off', // incompatible with most recent (attribute-form) generic types RFC
+        'no-unused-vars': ['error', {
+          args: 'none',
+          caughtErrors: 'none',
+          ignoreRestSiblings: true,
+          vars: 'all',
+          varsIgnorePattern: '^\\$\\$Props$',
+        }],
+
+        ...pluginSvelte.configs.recommended.rules as any,
```

Comment:
I wonder how many rules are in it? Should we just expand it inline so have a clear image what to include and compose (for example, would it include some rules in the stylistic category?)

Reply:
There are [14](https://github.com/sveltejs/eslint-plugin-svelte/blob/d22f6e957f00c2046f72e46d825727ad5ff2f658/src/configs/recommended.ts?plain=1#L11-L24) rules marked as recommended today, none of which are categorized as [stylistic](https://sveltejs.github.io/eslint-plugin-svelte/rules/#stylistic-issues).

I don't have an opinion on this issue, we can be more explicit by expanding the rules.

Reply:
expanded: https://github.com/antfu/eslint-config/pull/364/commits/9e31d9d48ebfc7bc463881f826e4e8f27155a234